src/main/java/org/folio/entlinks/service/authority/AuthoritySourceFileService.java

```diff
@@ -252,16 +236,38 @@ private void copyModifiableFields(AuthoritySourceFile existingEntity, AuthorityS
     }
   }
 
-  private void updateSequenceStartNumber(AuthoritySourceFile existing, AuthoritySourceFile modified) {
-    if (Objects.equals(existing.getHridStartNumber(), modified.getHridStartNumber())) {
-      return;
-    }
+  public void createSequence(String sequenceName, int startNumber) {
+    var command = String.format("""
+            CREATE SEQUENCE %s MINVALUE %d INCREMENT BY 1 OWNED BY %s.authority_source_file.sequence_name;
+            """,
+        sequenceName, startNumber, moduleMetadata.getDBSchemaName(folioExecutionContext.getTenantId()));
+    jdbcTemplate.execute(command);
+  }
 
-    var sequenceName = existing.getSequenceName();
-    var startNumber = (int) modified.getHridStartNumber();
+  public void deleteSequence(String sequenceName) {
+    var command = String.format("DROP SEQUENCE IF EXISTS %s.%s;",
+        moduleMetadata.getDBSchemaName(folioExecutionContext.getTenantId()), sequenceName);
+    jdbcTemplate.execute(command);
+  }
+
+  private void updateSequenceStartNumber(String sequenceName, int modifiedStartNumber) {
     var command = String.format("ALTER SEQUENCE %s RESTART WITH %d OWNED BY %s.authority_source_file.sequence_name;",
-      sequenceName, startNumber, moduleMetadata.getDBSchemaName(folioExecutionContext.getTenantId()));
+        sequenceName, modifiedStartNumber, moduleMetadata.getDBSchemaName(folioExecutionContext.getTenantId()));
     jdbcTemplate.execute(command);
   }
 
+  private void updateSequenceStartNumber(AuthoritySourceFile existing, AuthoritySourceFile modified) {
+    if (!Objects.equals(existing.getHridStartNumber(), modified.getHridStartNumber())
+        && existing.getHridStartNumber() != null && modified.getHridStartNumber() != null) {
+      var sequenceName = existing.getSequenceName();
+      var modifiedStartNumber = (int) modified.getHridStartNumber();
+      var existingStartNumber = (int) existing.getHridStartNumber();
+      if (modifiedStartNumber > existingStartNumber) {
+        updateSequenceStartNumber(sequenceName, modifiedStartNumber);
+      } else {
+        deleteSequence(sequenceName);
+        createSequence(sequenceName, modifiedStartNumber);
```

Comment:
Can we do it in all cases when we need to change sequence? A more additional logic in code we have - more bugs could occure. So, I think will be better to do recreate each case.